lib/srv/alpnproxy/local_proxy.go

```diff
@@ -331,6 +339,21 @@ func (l *LocalProxy) StartHTTPAccessProxy(ctx context.Context) error {
 	return nil
 }
 
+func writeKubeError(rw http.ResponseWriter, kubeError *apierrors.StatusError, logger logrus.FieldLogger) {
+	errorBytes, err := json.Marshal(kubeError.Status())
```

Comment:
don't use json.Marshal, instead use the native serializer

```go
var (
	// kubeScheme is the runtime Scheme that holds information about supported
	// message types.
	kubeScheme = runtime.NewScheme()
	// kubeCodecs creates a serializer/deserizalier for the different codecs
	// supported by the Kubernetes API.
	kubeCodecs = serializer.NewCodecFactory(kubeScheme)
)
	data, err := runtime.Encode(kubeCodecs.LegacyCodec(), status)
	if err != nil {
		
		trace.WriteError(rw, respErr)
		return
	}
	rw.Header().Set(responsewriters.ContentTypeHeader, "application/json")
	// Always write the correct error code in the response so kubectl can parse
	// it correctly. If response code and status.Code drift, kubectl prints
	// `Error from server (InternalError): an error on the server ("unknown")
	// has prevented the request from succeeding`` instead of the correct reason.
	rw.WriteHeader(trace.ErrorToCode(respErr))
	if _, err := rw.Write(data); err != nil {
		f.log.Warningf("Failed writing kube error response body: %v", err)
	}
```